contractcourt/chain_arbitrator.go

```diff
@@ -137,10 +137,9 @@ type ChainArbitratorConfig struct {
 	// Sweeper allows resolvers to sweep their final outputs.
 	Sweeper *sweep.UtxoSweeper
 
-	// SettleInvoice attempts to settle an existing invoice on-chain with
-	// the given payment hash. ErrInvoiceNotFound is returned if an invoice
-	// is not found.
-	SettleInvoice func(lntypes.Hash, lnwire.MilliSatoshi) error
+	// Registry is the invoice database that is used by resolvers to lookup
+	// preimages and settle invoices.
+	Registry *invoices.InvoiceRegistry
```

Comment:
This interface already exists as `htlcswitch.InvoiceDatabase`. In a previous iteration, I used that in `chain_arbitrator.go`. But in order to do so, the interface needed to be moved outside of `htlcswitch` to prevent a circular dependency. There was some objection to that, so I took it out again.

Added the commit back in now (`invoices+htlcswitch: move registry interface`). I am in favor of creating the decoupling in that way. Let me know if you see it differently. If ok, I can squash the commit.

Reply:
Btw, we can't just have the `SettleInvoice` closure, because we use two methods from `InvoiceRegistry`.

Reply:
Will check out the commit, but an advantage of introducing a slimmer/similar interface is that the package expresses to outside packages its requirements, and doesn't need to import another package to express its interface requirements. 

Yeah, it would be more than just `SettleInvoice`: it would also be the additional methods it needs as well.